tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,615 @@
+					storageClass.Name))
+				sourceScNameList = append(sourceScNameList, storageClass.Name)
+			}
+
+			err := SetDestinationKubeConfig()
+			dash.VerifyFatal(err, nil, "Setting destination kubeconfig")
+
+			for _, sc := range sourceScNameList {
+				scName = fmt.Sprintf("replica-sc-%v", RandomString(3))
+				v1obj := metaV1.ObjectMeta{
+					Name: scName,
+				}
+				reclaimPolicyDelete := v1.PersistentVolumeReclaimDelete
+				bindMode := storageApi.VolumeBindingImmediate
+				scObj := storageApi.StorageClass{
+					ObjectMeta:        v1obj,
+					Provisioner:       k8s.CsiProvisioner,
+					Parameters:        params,
+					ReclaimPolicy:     &reclaimPolicyDelete,
+					VolumeBindingMode: &bindMode,
+				}
+				_, err = storage.Instance().CreateStorageClass(&scObj)
+				log.FailOnError(err, "Creating sc on dest cluster")
+				storageClassMapping[sc] = scName
+			}
+
+			err = SetSourceKubeConfig()
+			dash.VerifyFatal(err, nil, "Setting source kubeconfig")
+		})
+
+		Step("Default restore of applications by replacing the existing resources with NS level PSA", func() {
+			log.InfoD("Default restore of applications by replacing the existing resources with NS level PSA")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			var mutex sync.Mutex
+			for _, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer wg.Done()
+					defer GinkgoRecover()
+					defaultRestoreName := fmt.Sprintf("%s-%s-default", RestoreNamePrefix, backupName)
+					err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, make(map[string]string), SourceClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete, scheduledAppContexts)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating default restore for backup with replace policy [%s]", defaultRestoreName))
+					mutex.Lock()
+					restoreNames = append(restoreNames, defaultRestoreName)
+					mutex.Unlock()
+				}(backupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Restore of applications with NS and StorageClass mapping with NS level PSA", func() {
+			log.InfoD("Restore of applications with NS and StorageClass mapping with NS level PSA")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			var mutex sync.Mutex
+			for i, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer wg.Done()
+					defer GinkgoRecover()
+					customRestoreName := fmt.Sprintf("%s-%s-custom-ns-sc", RestoreNamePrefix, backupName)
+					mutex.Lock()
+					namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored-1"
+					mutex.Unlock()
+
+					err = CreateRestoreWithReplacePolicyWithValidation(customRestoreName, backupName, namespaceMapping, DestinationClusterName, BackupOrgID, ctx, storageClassMapping, ReplacePolicyDelete, scheduledAppContexts)
```

Comment:
Need to check if the validation goes fine in case of parallel restore otherwise we need to add it as seperate step
@snigdha-px

Reply:
Default restore goes through other needs to be sequential .